tests/test_attributes.py

```diff
@@ -72,18 +72,22 @@ def test_attributes(browser, netid, utils, sp_url, sp_domain, secrets, test_env)
 
         # eduPersonScopedAffiliation is a unique case
         wanted_key = 'eduPersonScopedAffiliation'
-        result = list(filter(lambda x: x.startswith(wanted_key), attribute_data))
-        found_values = result[0].split("= ", 1)[1].split(";")
+
+        result = next(filter(lambda x: x.startswith(wanted_key), attribute_data))
+        found_values = set(result.split("= ")[1].split(";"))
 
         for key, value in attributes_to_test.items():
-            # matches on exact string only. mail won't count as a match for uwEduEmail and uwEduEmail won't match mail
-            if key != 'eduPersonScopedAffiliation':
-                assert (f'{key} = {value}' in attribute_data)
-            # eduPersonScopedAffiliation values may not be in the same order every time.
-            elif key == 'eduPersonScopedAffiliation':
-                assert len(found_values) == 4
-                assert 'employee@washington.edu' in found_values
-                assert 'student@washington.edu' in found_values
-                assert 'staff@washington.edu' in found_values
-                assert 'member@washington.edu' in found_values
+            if key == wanted_key:
+                print(f'found_values {found_values}')
```

Comment:
(Non-blocking): I think it's good you want to include the found_values output, so that we know if they change. However, using print statements in tests is generally not a good idea, because without certain environment settings, `print` output is not guaranteed to be seen, and also provides no traceback information.

In this case, I think the "best" solution would be to simply dump the contents in the event that the assertion fails (which is the only time we need to care), as that may indicate we need to update our tests (as you pointed out in another comment). You can achieve this by providing the error messasge as a second arugment of the assert statement, like so:

```python
assert found_values == {
   ...
}, f'Unexpected affiliation attributes: {found_values}'
```

https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement

> The extended form, assert expression1, expression2, is equivalent to
`[...] if not expression1: raise AssertionError(expression2)`

Reply:
Ha, I left that in there by accident as I was getting the test tuned.
I am going to integrate that, something about happy accidents 👍